scripts/west_commands/runners/intel_adsp.py

```diff
@@ -48,6 +50,10 @@ def __init__(self,
         else:
             self.key = os.path.join(DEFAULT_KEY_DIR, default_key)
 
+        self.tool_opt = []
+        for opts in [shlex.split(opt) for opt in tool_opt]:
+            self.tool_opt += opts
```

Comment:
This looks wrong to me, I suspect it leads to the situation already mentioned:
```
west flash --remote-host=192.168.0.1 --tool-opt="-r 12345 --arg='white space'"

=> tool_opt_args = [   "-r",    "12345",   "--arg='white",   "space'" ] # BAD
```
Did you try this?

I think you should remove this code and ask the users to do this instead:
```
west flash --remote-host=192.168.0.1 --tool-opt=-r --tool-opt=12345 --tool-opt=--arg='white space'

=> tool_opt_args = [   "-r",    "12345",   "--arg=white space" ] # GOOD
```
It is important to do this right considering the multiple levels of nesting: twister->west->cavstool

Reply:
Hi @marc-hb , I think I fixed according to the suggestion from you and @mbolivar-nordic , could you please take a look again? thanks!